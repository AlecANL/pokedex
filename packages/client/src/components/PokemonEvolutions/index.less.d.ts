// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'image-container': string;
  'pokemon-image': string;
  'pokemon-image-container': string;
  'text-container': string;
}
declare var cssExports: CssExports;
export = cssExports;
