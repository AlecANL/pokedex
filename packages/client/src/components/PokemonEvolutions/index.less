.pokemon-image-container {
  display: flex;
  padding-top: 10px;
  padding-bottom: 10px;
  .pokemon-image {
    display: flex;
    flex-direction: column;
    max-width: 110px;
    margin-right: 10px;

    .image-container {
      img {
        width: 100%;
      }
    }

    .text-container {
      text-align: center;
      font-size: 15px;
    }
  }
}
