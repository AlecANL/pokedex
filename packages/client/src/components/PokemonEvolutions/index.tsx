import React from 'react';
import Panel from '../Panel';
import { formatMessage } from 'umi-plugin-react/locale';
import style from './index.less';
import { zeroPad } from '@/helpers';

interface IPokemonEvolutions {
  /**
   * Pokemon evolutions array
   */
  evolutions: IEvolution[];
}

/**
 * This component allow to show pokemon evolutions. Informations are encapuslated in a <Panel/> component
 * @param props
 */
const PokemonEvolutions: React.FC<IPokemonEvolutions> = props => {
  const { evolutions } = props;
  return (
    <Panel
      title={formatMessage({ id: 'pages.evolutions' })}
      content={
        <div className={style['pokemon-image-container']}>
          {evolutions.map(evolution => {
            return (
              <div className={style['pokemon-image']}>
                <div className={style['image-container']}>
                  <img
                    src={`https://assets.pokemon.com/assets/cms2/img/pokedex/detail/${zeroPad(
                      evolution.id,
                      3,
                    )}.png`}
                  />
                </div>
                <div className={style['text-container']}>
                  <span>{evolution.name}</span>
                </div>
              </div>
            );
          })}
        </div>
      }
    />
  );
};

export default PokemonEvolutions;
