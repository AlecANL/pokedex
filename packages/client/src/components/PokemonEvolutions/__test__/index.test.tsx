import React from 'react';
import renderer from 'react-test-renderer';
import PokemonEvolutions from '..';

jest.mock('umi-plugin-react/locale');

describe('Components/PokemonEvolutions', () => {
  it('Get snapshot', () => {
    const component = renderer.create(<PokemonEvolutions evolutions={[]} />);
    let snap = component.toJSON();
    expect(snap).toMatchSnapshot();
  });
});
