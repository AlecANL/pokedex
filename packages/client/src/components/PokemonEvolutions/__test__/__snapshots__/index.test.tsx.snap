// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/PokemonEvolutions Get snapshot 1`] = `
<div
  className="panel-component"
>
  <div>
    <div
      className="pokemon-image-container"
    />
  </div>
</div>
`;
